transport/pool.go

```diff
@@ -0,0 +1,252 @@
+package transport
+
+import (
+	"errors"
+	"math"
+	"math/rand"
+	"runtime"
+	"time"
+
+	"scylla-go-driver/frame"
+	. "scylla-go-driver/frame/request"
+	. "scylla-go-driver/frame/response"
+
+	"go.uber.org/atomic"
+)
+
+// Node Connection Pool - One per node, Node struct holds pointer to it
+// This struct holds connection to shards along with helper variables.
+type ConnPool struct {
+	conns     []atomic.Value
+	alertChan chan uint16
+
+	nrShards  uint16
+	msbIgnore uint8
+
+	idx atomic.Uint64
```

Comment:
Grouping is not needed and conns should be close to idx imo.
Can we have it like G1,G0,G2?